src/NServiceBus.Core/Serializers/XML/XmlMessageSerializer.cs

```diff
@@ -70,9 +70,10 @@ public object[] Deserialize(Stream stream, IList<Type> messageTypesToDeserialize
         /// </summary>
         public void Serialize(object message, Stream stream)
         {
-            var serializer = new Serializer(mapper, conventions, cache, SkipWrappingRawXml, Namespace);
-            var buffer = serializer.Serialize(message);
-            stream.Write(buffer, 0, buffer.Length);
+            using (var serializer = new Serializer(mapper, stream, message, conventions, cache, SkipWrappingRawXml, Namespace))
+            {
+                serializer.Serialize();
```

Comment:
Note: I'm personally always unsure what the better design is regarding what comes into the ctor and what into the method. In the serializer case I would tend to add everything which is cacheable, bound to the lifetime of the object etc. into the ctor and everything which cannot be reused over multiple calls into the method. So I would tend to add the StreamWriter into the method not the ctor.  @SimonCropp How do you approach those decisions?